src/main/java/org/wikivoyage/listings/Main.java

```diff
@@ -1,252 +1,32 @@
-        Iterable<Listing> listings = new ListingsIterable(dumpPath);
-        
-        // Write listings to intermediate file.
-        if (useIntermediateFile) {
-            log.info("Write intermediate file with parsed listings");
-            String javaSerialFile = fileNames.workingDirPath("serialized-pois.bin");
-            FileUtils.removeFile(javaSerialFile);
-            new JavaSerializedObject().write(listings, javaSerialFile, dumpDate);
-            listings = new JavaSerializedIterable(javaSerialFile);
-        }
-        listings = validate(listings);
-        // Write listings to all the output formats.
-        for (OutputFormat format: formats.values()) {
-            writeFormat(listings, language, dumpDate, latestDumpDate, format);
-        }
-    }
-    
-    private static void writeFormat(Iterable<Listing> listings, String language,
-            String dumpDate, String latestDumpDate, OutputFormat format) throws FileUtilsException {
-        log.info(
-                "Write output file for language " + language +
-                ", dump " + dumpDate + ", format " + format.getDefaultExtension().substring(1)
-            );
-        String fileName = fileNames.getListingPath(language, dumpDate, format.getDefaultExtension(), false);
-        try {
-            format.write(listings, fileName, dumpDate);
-            if (dumpDate.equals(latestDumpDate)) {
-                String latestFileName = fileNames.getListingPath(
-                        language, "latest", format.getDefaultExtension(), false
-                );
-                FileUtils.removeFile(latestFileName);
-                FileUtils.copyFile(fileName, latestFileName);
-            }
-            String fileNameArchive = fileNames.getListingPath(language, dumpDate, format.getDefaultExtension(), true);
-            FileUtils.archive(fileName, fileNameArchive);
-        } catch (WriteOutputException e) {
-            System.out.println("Failed to write file: " + e.getMessage());
-        }
+        Main main = new Main();
+        main.parseAndRun(args);
     }
 
-    private static void createListingsDir() throws FileUtilsException
-    {
-        FileUtils.createDirectory(fileNames.getListingsDir());
-    }
+    private void parseAndRun(String[] args) {
+        // Parse CLI
+        CommandLine cli = new CommandLine();
+        cli.parse(args);
 
-    private static void createDumpsCacheDir() throws FileUtilsException
-    {
-        FileUtils.createDirectory(fileNames.getDumpsCacheDir());
-    }
+        // Create project structure in FileSystem
+        ProjectFolders folders = new ProjectFolders(cli.listingsDir, cli.dumpsCacheDir, cli.workingDir);
+        folders.createProjectFolders();
 
-    private static void createWorkingDir() throws FileUtilsException
-    {
-        FileUtils.createDirectory(fileNames.getWorkingDir());
-    }
+        System.out.println();
+        System.out.println("CHRIS CONNOLLY RUNNING THIS");
```

Comment:
I guess you forgot to remove this?

Reply:
Yep, sorry.

Reply:
Just push a new commit to the same branch and it will magically be added to your PR :-)

Reply:
"the same branch" = master for you. By the way, the usual workflow in open source is this: 1) Post an issue describing a specific problem and how you will solve it 2) Create a branch with a name that refers to this issue and fixes only that specific issue 3) Send pull request. You can do this for several issues in parallel, and the maintainers can decide what issues are real issues or non-issues, and prioritize what branch to review and pull first.
Cheers!

Reply:
Thanks for that information. I've dealt with issues on Kanban boards etc but working with open source is new to me :) Cheers!